File: addToguether.ts
const addToguether = (first: number) => {
  validateNumber(first, "first")

  return (second: number) => {
    validateNumber(first, "second")
    return first + second
  }
}

const validateNumber = (number: number, position: string) => {
  if (typeof number !== "number" || Number.isNaN(number))
    throw new Error(`${position} must be a valid number`)
}


console.log(addToguether(2)(30))

File: addToguether.py
def add_together(first: float):
    validate_number(first, "First")

    def inner(second: float) -> float:
        validate_number(second, "Second")
        return first + second

    return inner


def validate_number(number: float, position: str):
    if not isinstance(number, (int, float)) or isinstance(number, bool):
        raise ValueError(f"{position} must be  a valid number")


# Example usage
print(add_together(2)(30))  # Output: 32
# These will raise ValueError:
#add_together("not a number")(2)
# add_together(2)("not a number")
#add_together(True)(2)  # Boolean values are not allowed
